Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,311 +14,403 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
-
-#include "Foundation/NSOrderedSet.h"
+#import <VAListHelper.h>
 
 @implementation NSOrderedSet
+
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSet {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] init] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [NSOrderedSet orderedSetWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
```

Comment:
Your factory methods should use `self` instead of `NSOrderedSet`. Otherwise, `[NSMutableOrderedSet orderedSet]` and friends will return *immutable* ones.